packages/extension-ui/src/partials/Name.tsx

```diff
@@ -14,15 +14,16 @@ interface Props {
   label?: string;
   onBlur?: () => void;
   onChange: (name: string | null) => void;
+  value?: string;
 }
 
-export default function Name ({ address, className, isFocused, label, onBlur, onChange }: Props): React.ReactElement<Props> {
+export default function Name ({ address, className, isFocused, label, onBlur, onChange, value }: Props): React.ReactElement<Props> {
   const { t } = useTranslation();
   const { accounts } = useContext(AccountContext);
   const isNameValid = useMemo(() => isNotShorterThan(3, t<string>('Account name is too short')), [t]);
 
   const account = accounts.find((account) => account.address === address);
-  const startValue = account && account.name;
+  const startValue = value || account?.name;
```

Comment:
This is the issue for the comment above - the above component only sets the value once. Here it is not available on the first render, so it never gets used.

Reply:
Not sure I understand. Here `value` comes from the props and every time this changes, the `Name` component rerenders, and `startValue` will get a new value.

If I remove the `useEffect` from `ValidatedInput` and log both `startValue` from `Name` and `defaultValue` from `ValidatedInput`, then I get:
```
startValue undefined
defaultValue undefined
startValue root
defaultValue root
```

the new value, with "root" does end up in the `ValidatedInput`, but it doesn't re-render :man_shrugging:

Reply:
No, it doesn't re-create, it just passes props to the component.

In that component you only set the initial state right on creation.